subsys/net/ip/net_core.c

```diff
@@ -64,11 +64,16 @@ LOG_MODULE_REGISTER(net_core, CONFIG_NET_CORE_LOG_LEVEL);
 #include "net_stats.h"
 
 #if defined(CONFIG_NET_NATIVE)
-static inline enum net_verdict process_data(struct net_pkt *pkt,
-					    bool is_loopback)
+static inline enum net_verdict process_data(struct net_pkt *pkt)
 {
 	int ret;
-	bool locally_routed = false;
+
+	/* If there is no data, then drop the packet. */
+	if (!pkt->frags) {
```

Comment:
```suggestion
	if (pkt->frags == NULL) {
```

Reply:
@pdgendgt it's a coding guideline issue so you may make it an actual request for change

"The controlling expression of an if statement and the controlling expression of an iteration-statement shall have essentially Boolean type"
 https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html#:~:text=The%20controlling%20expression%20of%20an%20if%20statement%20and%20the%20controlling%20expression%20of%20an%20iteration%2Dstatement%20shall%20have%20essentially%20Boolean%20type

Reply:
I know, but it bothers me too to keep making these comments 😅 maybe we need clang tidy in `check_compliance.py` with the [`readability-implicit-bool-conversion`](https://clang.llvm.org/extra/clang-tidy/checks/readability/implicit-bool-conversion.html) rule enabled.

Reply:
+1 to that - didn't know it existed and was such a low hanging fruit

Reply:
_Sorry, edited my previous message._
Not really low hanging fruit IMO, clang-tidy requires a compilation database to run properly, so it can only be done after a build.

Reply:
> _Sorry, edited my previous message._ Not really low hanging fruit IMO, clang-tidy requires a compilation database to run properly, so it can only be done after a build.

Ahh - I read too quickly, thought you meant clang-**format**. That makes a lot more sense now :)